### PR TITLE
Feature frame from recordbatch

### PR DESCRIPTION
**Pull request by ErikDeSmedt**
This pull request provides a way to create a `polars::frame::DataFrame` from `Vec<arrow::record_batch::RecordBatch>`.

```
let batches : Vec<RecordBatch> = ...;
let df = DataFrame::try_from(batches).unwrap();
```

### CODE REVIEW

File: polars/src/series/mod.rs [hunk 1]
```diff
@@ -1145,6 +1179,12 @@ mod test {
         ca.into_series();
     }
 
+    #[test]
+    fn new_series_from_arrow_primitive_array() {
+        let array = UInt64Array::from(vec![1, 2, 3, 4, 5]);
+        let _array_ref: ArrayRef = Arc::new(array);
```
Comment (ritchie46) [inline polars/src/series/mod.rs:1185]: unfinished test?

File: polars/src/frame/mod.rs [hunk 2]
```diff
@@ -1609,16 +1611,97 @@ impl Default for DataFrame {
...
+        for (field_idx, field) in Iterator::enumerate(schema.fields().iter()) {
+            let chunks: Vec<ArrayRef> = batches
+                .iter()
+                .map(|batch| batch.column(field_idx).clone())
+                .collect();
+            let series = Series::try_from((field.name().as_ref(), chunks))?;
+            series_vec.push(series);
+        }
+
+        DataFrame::new(series_vec)
```
Comment (ritchie46) [inline polars/src/frame/mod.rs:1651]: I believe recordbatches are guaranteed to have the same length and unique column names. If this is correct you can use `DataFrame::new_no_checks`. This skips the assertion of these requirements.

**Review by ritchie46 — CHANGES_REQUESTED** [review]
LGTM. I have a few minor remarks.

I think this can lead to some code deduplication in the `ser` module. There I also convert from record batch to Series/ DataFrame.

File: polars/src/frame/mod.rs (hunk 2 above)
Comment (ErikDeSmedt) [reply polars/src/frame/mod.rs:1651]: `RecordBatch` guarantees same length columns but not unique column names. Do you think using `DataFrame::new` is approriate or would you manually check for unique column names instead?

The snippet below shows that you can create a `RecordBatch` where both columns are named `"a"`
```rust
extern crate arrow;

#[cfg(test)]
mod tests {

    use std::sync::Arc;
    use arrow::datatypes::{DataType, Schema, Field};
    use arrow::record_batch::RecordBatch;
    use arrow::array::{Int64Array, Float64Array};

    #[test]
    fn it_works() {

        let schema = Arc::new(Schema::new(vec![
            Field::new("a", DataType::Float64, false),
            Field::new("a", DataType::Int64, false)
        ]));

        let arr1 = Arc::new(Float64Array::from(vec![1.0, 2.0, 3.0, 4.0, 5.0]));
        let arr2 = Arc::new(Int64Array::from(vec![1, 2, 3, 4, 5]));

        RecordBatch::try_new(
            schema,
            vec![arr1, arr2]
        ).expect("Can create batch with same column name");

    }
}
```

File: polars/src/series/mod.rs (hunk 1 above)
Comment (ErikDeSmedt) [reply polars/src/series/mod.rs:1185]: Thanks for you quick review.

There is indeed some duplication with `src\frame\ser\mod.rs`. I'll look into that later this week.

I missed a part of the test and plan to shoot in the same commit.

File: polars/src/frame/mod.rs (hunk 2 above)
Comment (ritchie46) [reply polars/src/frame/mod.rs:1651]: Then the DataFrame creation as is, is best indeed. :+1:

**Pull request merged**